### PR TITLE
Different improvements, regarding final modifiers, try with resources, and other

### PR DESCRIPTION
**Pull request by alxkm**
There several of different improvements. 
Regarding final modifiers 
Try with resources 
Lambda reference 
And other

### CODE REVIEW

**Comment by tzolov** [comment]
Thank you @alxkm 
Could you please rebase (not merge) your PR to of upstream/main and resolve the conflicts. 
E.g. `git pull -r upstream main `

File: spring-ai-core/src/main/java/org/springframework/ai/ResourceUtils.java [hunk 1]
```diff
@@ -25,6 +25,10 @@
  */
 public class ResourceUtils {
 
+	private ResourceUtils() {
```
Comment (markpollack) [inline spring-ai-core/src/main/java/org/springframework/ai/ResourceUtils.java:28]: Spring conventions are to make the class abstract, e.g. StringUtils

File: spring-ai-core/src/main/java/org/springframework/ai/aot/AiRuntimeHints.java [hunk 2]
```diff
@@ -40,6 +40,10 @@
  */
 public class AiRuntimeHints {
 
+	private AiRuntimeHints() {
```
Comment (markpollack) [inline spring-ai-core/src/main/java/org/springframework/ai/aot/AiRuntimeHints.java:43]: Similar to before, will create as abstract.  Likely should change the name to RuntimeHintUtils, but that will be a big commit, so will do another time.

File: spring-ai-core/src/main/java/org/springframework/ai/document/Document.java [hunk 3]
```diff
@@ -203,7 +203,7 @@ else if (!content.equals(other.content))
 
 	@Override
 	public String toString() {
-		return "Document{" + "id='" + id + '\'' + ", metadata=" + metadata + ", content='" + new String(content) + '\''
+		return "Document{" + "id='" + id + '\'' + ", metadata=" + metadata + ", content='" + content + '\''
```
Comment (markpollack) [inline spring-ai-core/src/main/java/org/springframework/ai/document/Document.java:206]: regenerated toString to also include media

**Comment by markpollack** [comment]
refined and merged in cb6662a61e421af43801c4090e3330d2b1811edf

thanks!

**Pull request closed without merging**